### ISSUE TITLE
Adopt the usethis model of the "current project"

### ISSUE THREAD

**Issue by pat-s** [ropenscilabs/travis#124]
Can we adopt the usethis model of the "current project" here (after merging this PR)?

_Originally posted by @krlmlr in https://github.com/ropenscilabs/travis/pull/123_

By using `package = NULL` as the first arg instead of `path = usethis::proj_get()`.

**Comment by krlmlr** [comment]
Can we stop exposing the `path` argument altogether, and not expose a `package` argument?

**Comment by pat-s** [comment]
So you want to harcode using the current dir when calling `github_info()` and friends?
We could do so and put a note in the help page.

Then the only {usethis} dep left is `github_token()`.
This we could also implement ourselves 

```r
    token <- Sys.getenv("GITHUB_PAT", "")
    if (token == "") 
        Sys.getenv("GITHUB_TOKEN", "")
    else token
```

and then we can get rid of the usethis IMPORT completely.

**Comment by krlmlr** [comment]
Not quite. I'd like to use `usethis::proj_get()` to find the current project.

**Comment by pat-s** [comment]
> I'd like to use usethis::proj_get() to find the current project.

But without exposing an arg to change the repo dir manually?

**Comment by krlmlr** [comment]
Yes, that's the beauty of it.

**Comment by pat-s** [comment]
So just move `usethis::proj_get()` inside the fun body and remove the `path` arg? 
I might be overcomplicating this.

**Comment by krlmlr** [comment]
If this works, yes. We need to link prominently to `?proj_get` too.

**Comment by pat-s** [comment]
ATM `github_info()` are internal funs. Should we expose them?
You are only talking about `github_info()` and `github_repo()`, correct?

**Comment by krlmlr** [comment]
I'm talking about all currently exported functions. I'm not sure we need to export any new functions.

**Comment by pat-s** [comment]
I am not sure if were both on the same page here.

- `github_info()` and `github_repo()` are internal, so the change would not be user visible (which is not a problem)
- all other functions have the `repo` arg as the first one which expects a repo-slug. I would like to keep this one to keep the flexibility of specifying other repos without having to be in project root of this project. This is also used in the test suite heavily because repo autodetection does not work within the temporary dirs that `testthat` runs in

**Comment by krlmlr** [comment]
I propose to get rid of the `repo` argument and `usethis::with_project()` in the tests.

**Comment by pat-s** [comment]
Do you want to tackle this? I think I can hardly match all of your ideas in detail here and would focus on other open tasks instead - rather than having a long review process of a potential PR :)

**Comment by krlmlr** [comment]
This will touch code all over the entire repository. It's difficult to have other pull requests or branches open in parallel without running into conflicts, which I'd rather avoid.

**Comment by krlmlr** [comment]
What other open tasks do yo want to focus on?

**Comment by pat-s** [comment]
I was thinking that you'd do this after all PRs are merged. But simply assign this one to you and say that I won't do it then :)

> What other open tasks do yo want to focus on?

{tic}, so the review can proceed. I've also submitted {circle} for ropensci review which got blocked by the {tic} review, the same will happen to {travis}.

I want to get all of this done to switch focus :)
